python/cuml/cuml/internals/array_sparse.py

```diff
@@ -81,34 +77,38 @@ def __init__(
         convert_to_mem_type=None,
         convert_index=None,
         convert_format=True,
+        check_rows=None,
+        check_cols=None,
     ):
         if not isinstance(data, SparseCumlArrayInput):
-            if cpx_sparse.isspmatrix(data):
+            if cpx_sparse.issparse(data):
                 from_mem_type = MemoryType.device
-            elif scipy_sparse.isspmatrix(data):
+            elif scipy_sparse.issparse(data):
                 from_mem_type = MemoryType.host
             else:
                 raise ValueError(
-                    "A sparse matrix is expected as input. "
-                    "Received %s" % type(data)
+                    "A sparse matrix is expected as input. Received %s"
+                    % type(data)
                 )
 
-            if not isinstance(data, sparse_matrix_classes):
+            if not data.ndim == 2:
+                raise ValueError(
+                    "Expected 2D input, got input with shape {data.shape}"
+                )
+
+            if data.format != "csr":
                 if convert_format:
                     debug(
-                        "Received sparse matrix in {} format but CSR is "
-                        "expected. Data will be converted to CSR, but this "
-                        "will require additional memory copies. If this "
-                        "conversion is not desired, set "
-                        "set_convert_format=False to raise an exception "
-                        "instead.".format(type(data))
+                        "Received sparse matrix in {data.format!r} format "
```

Comment:
I think a note in the docstring describing that CSR is the optimal choice might be a good idea as well.

Reply:
That note already exists in the docstring and IMO should be sufficient. The change here was just moving an existing debug log to an f-string, though I did see when you commented that I'd missed the leading `f`. Updated!